### PR TITLE
do not run top-level lifecycle scripts in node_module generation

### LINKED ISSUE #12: npmBuild should default to `npm build`
Currently, `buildNpmPackage { src = ./.; }` also requires `npmBuild`, which has a sensible default in most cases. We should set it to `npm build`.

### PR DESCRIPTION
**Pull request by yorickvP**
Packages can contain [lifecycle scripts](https://docs.npmjs.com/misc/scripts), that `npm ci` will run during `node_module` generation. Since we are not building the package at that stage, that would be wrong. Let's strip them out of the `package.json` in this phase :).

The aim is that users of this repo should be able to use `npm pack` to produce the same output that we do (but packed in a tarball). Running the lifecycle scripts during our build sounds like a decent idea, since that is what the `npm ci && npm prune &&  npm pack` workflow would do outside of nix.

Fixes #12. This also means that it should not require any changes to the projects where it is used.

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -171,8 +175,9 @@ in rec {
       inherit name;
 
       configurePhase = ''
-        cp -r ${nodeModules}/node_modules ./node_modules
+        cp --reflink=auto -r ${nodeModules}/node_modules ./node_modules
```
Comment (zhenyavinogradov) [inline default.nix]: this change will be easier to understand if it is done in a separate commit

File: default.nix [hunk 2]
```diff
@@ -171,8 +175,9 @@ in rec {
...
         chmod -R u+w ./node_modules
+        patchShebangs .
```
Comment (zhenyavinogradov) [inline default.nix]: `./node_modules` is already patched, if you want to patch the sources do it before copying `node_modules` to avoid going through it again

**Comment by yorickvP** [comment]
Updated.
